lib/ex_aws/request.ex

```diff
@@ -104,6 +98,31 @@ defmodule ExAws.Request do
     end
   end
 
+  defp do_request(config, method, safe_url, req_body, full_headers, attempt) do
+    telemetry_event = Map.get(config, :telemetry_event, [:ex_aws, :request])
+    telemetry_options = Map.get(config, :telemetry_options, [])
+    telemetry_metadata = %{options: telemetry_options, attempt: attempt}
+
+    :telemetry.span(telemetry_event, telemetry_metadata, fn ->
+      config[:http_client].request(
+        method,
+        safe_url,
+        req_body,
+        full_headers,
+        Map.get(config, :http_opts, [])
+      )
+      |> case do
+        {:ok, %{status_code: status}} = result when status in 200..299 or status == 304 ->
+          telemetry_metadata = Map.put(telemetry_metadata, :result, :ok)
+          {result, telemetry_metadata}
+
+        result ->
+          telemetry_metadata = Map.put(telemetry_metadata, :result, :error)
+          {result, telemetry_metadata}
+      end
+    end)
+  end
```

Comment:
A minor stylistic thing - rather than duplicating the `Map.put` and result tuple construction in each case block, it might be a little cleaner to do something like:

```elixir
result = config[:http_client].request(...)
telemetry_result = case result do
  {:ok, %{status_code: status}} = result when status in 200..299 or status == 304 -> :ok
  _ -> :error
end

telemetry_metadata = Map.put(telemetry_metadata, :result, telemetry_result)
{result, telemetry_metadata}
```

Reply:
thanks for the tip :)